server/fishtest/templates/tests_view.mak

```diff
@@ -514,10 +518,41 @@
   document
     .getElementById("moon")
     .addEventListener("click", () => setHighlightTheme("dark"));
-</script>
 
-<script>
-  document.title = "${page_title} | Stockfish Testing";
+  async function downloadPGNs() {
+    try {
+      const response = await fetch(`/api/run_pgns/${run["_id"]}.pgn.gz`);
+      if (!response.ok) {
+        if (response.status === 404) {
+          alertError("No games found for this run");
+        } else {
+          alertError("Unable to download PGNs");
+        }
+        return;
+      }
+      const blob = await response.blob();
+
+      / Check if the blob is empty
+      if (blob.size === 0) {
+        alertError("No games found for this run");
```

Comment:
IMO the message is misleading. If the total size of the PGNs is zero (no games found) the API response is 404, so the blob size zero has another cause.

https://github.com/official-stockfish/fishtest/blob/0893ce71a4999953d1976fedae1aaaf0a39ee3b7/server/fishtest/rundb.py#L291-L298

https://github.com/official-stockfish/fishtest/blob/0893ce71a4999953d1976fedae1aaaf0a39ee3b7/server/fishtest/api.py#L541-L547

Reply:
Well I didn't get a blob size of zero, it was only error handling if the server condition was relaxed at some point.
I guess I can change the message.